drivers/include/rn2xx3.h

```diff
@@ -0,0 +1,699 @@
+    uint8_t tx_port;                   /**< Application TX port (between 1 and 223) */
+    uint8_t tx_mode;                   /**< TX mode, either confirmable of unconfirmable */
+    uint8_t rx_port;                   /**< RX port (between 1 and 223) */
+} loramac_settings_t;
+
+/**
+ * @brief   Configuration parameters for RN2483/RN2903 devices
+ */
+typedef struct {
+    uart_t uart;                       /**< UART interfaced the device is connected to */
+    uint32_t baudrate;                 /**< baudrate to use */
+    gpio_t pin_reset;                  /**< GPIO pin that is connected to the STATUS pin
+                                            set to GPIO_UNDEF if not used */
+    loramac_params_t loramac;          /**< loramac static configuration */
+} rn2xx3_params_t;
+
+/**
+ * @brief   RN2483/RN2903 device descriptor
+ */
+typedef struct {
+    netdev_t netdev;                   /**< Netdev parent struct */
+    /* device driver specific fields */
+    rn2xx3_params_t p;                 /**< configuration parameters */
+    loramac_settings_t loramac;        /**< loramac communication settings */
+
+    /* values for the UART TX state machine  */
+    char cmd_buf[RN2XX3_MAX_BUF];      /**< command to send data buffer */
+    mutex_t cmd_lock;                  /**< mutex to allow only one
+                                        *   command at a time */
+    uint8_t int_state;                 /**< current state of the device */
+
+    /* buffer and synchronization for command responses */
+    mutex_t resp_lock;                 /**< mutex for waiting for command
+                                        *   response */
+    char resp_buf[RN2XX3_MAX_BUF];     /**< command response data buffer */
+    uint16_t resp_size;                /**< counter for received char in response */
+    uint8_t resp_done;                 /**< check if response has completed */
+
+    /* buffer for RX messages */
+    char rx_tmp_buf[2];                /**< Temporary RX buffer used to convert
+                                        *   2 hex characters in one byte on the
+                                        *   fly. */
+    uint8_t rx_buf[RN2XX3_RX_MAX_BUF]; /**< RX data buffer */
+    uint16_t rx_size;                  /**< counter for received char in RX */
+    char rx_port[4];                   /**< RX port buffer */
+    uint8_t rx_port_size;              /**< port used for RX */
+
+    /* timers */
+    xtimer_t sleep_timer;              /**< Timer used to count module sleep time */
+    uint32_t sleep;                    /**< module sleep duration */
+} rn2xx3_t;
+
+/**
+ * @brief   Prepare the given RN2XX3 device
+ *
+ * @param[out] dev          RN2XX3 device to initialize
+ * @param[in]  params       parameters for device initialization
+ */
+void rn2xx3_setup(rn2xx3_t *dev, const rn2xx3_params_t *params);
+
+/**
+ * @brief   Initializes the RN2XX3 device
+ *
+ * @param[in] dev           RN2XX3 device descriptor
```

Comment:
isn't this an out param?

Reply:
not for `_init`, I used the out for `_setup`